tools/misc/calc_timeprop_spectrum

```diff
@@ -11,24 +11,29 @@ Fourier transform of DFTB+ TD data
 '''
 
 import sys
-import optparse
+import argparse
 import numpy as np
 from scipy import constants
 
-USAGE = """usage: %prog -d xx
+USAGE = """%(prog)s -d xx
 
 Reads output from TD calculation (after kick) and produces absorption spectra.
 
 Needs mux.dat, muy.dat and muz.dat files in working directory."""
 
 def main():
-    parser = optparse.OptionParser(usage=USAGE)
-    parser.add_option("-d", "--damping", action="store", dest="tau",
-                      help="damping constant in fs")
-    parser.add_option("-f", "--field", action="store", dest="field",
-                      help="field intensity in volts/angstrom")
+    parser = argparse.ArgumentParser(usage=USAGE)
+    parser.add_argument("-d", "--damping", dest="tau",
+                        help="damping constant in fs")
+    parser.add_argument("-f", "--field", dest="field",
+                        help="field intensity in volts/angstrom")
+
+    options, args = parser.parse_known_args()
+
+    if len(args) != 0:
+        parser.error("You must specify exactly two argument, the damping constant in fs and "
+                     "the field intensity in volts/angstrom")
```

Comment:
I know it's not your code-design, but since the damping constant and field intensity are required in any case, why make them optional in the first place?